packages/now-cli/src/util/pkg.ts

```diff
@@ -1,6 +1,6 @@
 import _pkg from '../../package.json';
 import { PackageJson } from '@vercel/build-utils';
 
-const pkg: PackageJson = _pkg;
+const pkg: PackageJson & typeof _pkg = _pkg;
```

Comment:
That's interesting, was TS complaining about something missing?

Reply:
It's kind of a cool optimization that I figured out. Because the base `PackageJson` type specifies all the properties as _optional_, it caused usage of this `pkg.ts` file to do type checking for the undefined case, even though we know the hard-coded values of the CLI `package.json` (name and version definitely do exist, for example). So adding this union type makes those hard-coded properties be non-optional so the checking elsewhere is not necessary.

The reason the base `PackageJson` type is being used at all is so that `pkg.dependencies` can be passed a dynamic property (`pkg.dependencies[name]` in a loop for example) because it defines as a map. _Without_ this base type, only the hard-coded values in the package.json could be specified and index syntax could not be used.

Reply:
Great description, thanks! 👍